plugins/module_utils/vm_snapshot.py

```diff
@@ -0,0 +1,188 @@
+        )
+
+    # This method is here for testing purposes!
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, VMSnapshot):
+            return NotImplemented
+        return all(
+            (
+                self.uuid == other.uuid,
+                self.domain_uuid == other.domain_uuid,
+                self.timestamp == other.timestamp,
+                self.label == other.label,
+                self.type == other.type,
+                self.automated_trigger_timestamp == other.automated_trigger_timestamp,
+                self.local_retain_until_timestamp == other.local_retain_until_timestamp,
+                self.remote_retain_until_timestamp
+                == other.remote_retain_until_timestamp,
+                self.block_count_diff_from_serial_number
+                == other.block_count_diff_from_serial_number,
+                self.replication == other.replication,
+            )
+        )
+
+    @classmethod
+    def get_by_uuid(
+        cls,
+        ansible_dict: Dict[Any, Any],
+        rest_client: RestClient,
+        must_exist: bool = False,
+    ) -> Optional[VMSnapshot]:
+        query = get_query(ansible_dict, "uuid", ansible_hypercore_map=dict(uuid="uuid"))
+        hypercore_dict = rest_client.get_record(
+            "/rest/v1/VirDomainSnapshot", query, must_exist=must_exist
+        )
+        vm_snapshot_from_hypercore = cls.from_hypercore(hypercore_dict)  # type: ignore
+        return vm_snapshot_from_hypercore
+
+    @classmethod
+    def get_by_snapshots_label(
+        cls,
+        label: str,
+        rest_client: RestClient,
+    ) -> List[Optional[TypedVMSnapshotToAnsible]]:
+        snapshots = [
+            cls.from_hypercore(hypercore_data=hypercore_dict).to_ansible()  # type: ignore
+            for hypercore_dict in rest_client.list_records(
+                "/rest/v1/VirDomainSnapshot", {"label": label}
+            )
+        ]
+
+        return snapshots
+
+    @classmethod
+    def get_by_snapshots_serial(
+            cls,
+            serial: str,
+            rest_client: RestClient,
+    ) -> List[Optional[TypedVMSnapshotToAnsible]]:
+        snapshots = [
+            cls.from_hypercore(hypercore_data=hypercore_dict).to_ansible()  # type: ignore
+            for hypercore_dict in rest_client.list_records(
+                "/rest/v1/VirDomainSnapshot", {"block_count_diff_from_serial_number": serial}
+            )
+        ]
+
+        return snapshots
+
+    @classmethod
+    def get_all_snapshots(
+            cls,
+            rest_client: RestClient,
+    ) -> List[Optional[TypedVMSnapshotToAnsible]]:
+        state = [
+            cls.from_hypercore(hypercore_data=hypercore_dict).to_ansible()  # type: ignore
+            for hypercore_dict in rest_client.list_records(
+                "/rest/v1/VirDomainSnapshot"
+            )
+        ]
+
+        return state
+
+    @classmethod
+    def get_snapshots_by_query(
```

Comment:
Hm, all `get_snapshots_by_query`, `get_all_snapshots`, `get_by_snapshots_serial`, `get_by_snapshots_label` - they are all the same. Only `get_by_uuid` is special - it does get via HTTP only one snapshot, not all.

Can you use get_snapshots_by_query in all of them, or maybe throw others (`get_by_snapshots_serial` etc away)?

Reply:
I will keep ``get_snapshots_by_query`` and throw others away.